### PR TITLE
Task/jenkins 44593 react 154

### PR DESCRIPTION
**Pull request by cliffmeyers**
# Description
- Upgrading from 15.3 to 15.4 as first step toward 16, and so that react-beautiful-dnd can be used for https://issues.jenkins-ci.org/browse/JENKINS-38323
- See [JENKINS-44593](https://issues.jenkins-ci.org/browse/JENKINS-44593).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by cliffmeyers** [comment]
Build of blueocean-core-js is currently failing in CI, investigating...

**Comment by cliffmeyers** [comment]
I'll need to resolve shrink conflicts again, but this one passed ATH on the first try, which is by all accounts a miracle of biblical proportions.

**Comment by michaelneale** [comment]
WOAH @cliffmeyers FIRST TRY!
hahah ok. Any impact on JDL @alvarolobato @scherler down the track?

**Comment by cliffmeyers** [comment]
Discussed with @alvarolobato and @scherler earlier today... @alvarolobato gave thumbs up for React upgrade provided it rolls into Blue Ocean 1.4 which was the original plan. This will not affect Blue Ocean 1.3 obviously.

**Comment by michaelneale** [comment]
ok cool!

**Comment by sophistifunk** [comment]
Was this by hand, or with the magic tools from FB?

File: Dockerfile.build [hunk 1]
```diff
@@ -43,7 +43,7 @@ RUN wget --no-verbose https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION
 RUN tar -C /usr/local --strip-components 1 -xJf /opt/nodejs.tar.xz
 RUN mkdir /.npm && chmod 777 /.npm
 # Update to latest version so optional dependancies + shrinkwrap work
-RUN npm install -g npm@4.3.0
+RUN npm install -g npm@3.10.3
```
Comment (michaelneale) [inline Dockerfile.build]: ??

**Comment by michaelneale** [comment]
downgrade of npm is troubling - is that needed?

**Comment by michaelneale** [comment]
ah - just making it consistent ;)

:bee:

**Comment by cliffmeyers** [comment]
@sophistifunk by hand. There were no code mods to make. This is me typing "npm install ..." like 20-30 times into terminal. Awful.

**Comment by sophistifunk** [comment]
horror

**Pull request merged**